src/app/fdctl/run/tiles/fd_sign.c

```diff
@@ -72,32 +72,24 @@ during_frag( void * _ctx,
   fd_sign_ctx_t * ctx = (fd_sign_ctx_t *)_ctx;
   FD_TEST( in_idx<MAX_IN );
 
-  switch( ctx->in_role[ in_idx ] ) {
-    case FD_KEYGUARD_ROLE_LEADER:
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], 32UL );
-      break;
-    case FD_KEYGUARD_ROLE_TLS:
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], 130UL );
-      break;
-    case FD_KEYGUARD_ROLE_GOSSIP:
-      if( sz>FD_KEYGUARD_SIGN_REQ_MTU ) {
-        FD_LOG_WARNING(("Corrupt gossip signing message with size %lu", sz));
-        *opt_filter = 1;
-        return;
-      }
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], sz );
-      break;
-    case FD_KEYGUARD_ROLE_REPAIR:
-      if( sz>FD_KEYGUARD_SIGN_REQ_MTU ) {
-        FD_LOG_WARNING(("Corrupt repair signing message with size %lu", sz));
-        *opt_filter = 1;
-        return;
-      }
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], sz );
-      break;
-    default:
-      FD_LOG_CRIT(( "unexpected link role %lu", ctx->in_role[ in_idx ] ));
+  int role = ctx->in_role[ in_idx ];
+
+  switch( role ) {
+  case FD_KEYGUARD_ROLE_LEADER:
+  case FD_KEYGUARD_ROLE_TLS:
+  case FD_KEYGUARD_ROLE_GOSSIP:
+  case FD_KEYGUARD_ROLE_REPAIR:
+    break;
+  default:
+    FD_LOG_CRIT(( "unexpected link role %d", role ));
+  }
+
+  if( sz>FD_KEYGUARD_SIGN_REQ_MTU ) {
+    FD_LOG_WARNING(( "oversz signing request (role=%d sz=%lu)", role, sz ));
```

Comment:
Again, just because of keyguard I think we should abort here. This points to a programmer error in the caller, which is one more bug away from swiping the key.